### PR TITLE
Fix alpha conversion CRD e2e's

### PR DESCRIPTION
**Pull request by liggitt**
**What type of PR is this?**
/kind failing-test

**What this PR does / why we need it**:
Fixes alpha e2e CRDs

**Which issue(s) this PR fixes**:
xref https://github.com/kubernetes/kubernetes/issues/78087

**Does this PR introduce a user-facing change?**:
```release-note
NONE
```

### CODE REVIEW

**Comment by liggitt** [comment]
cc @sttts @roycaihw 

The TMP commit enables the features and removes the e2e tags so we can see if the tests pass in presubmits. I will remove that commit before merge.

**Comment by roycaihw** [comment]
you could use `/test pull-kubernetes-e2e-gce-alpha-features` to trigger the non-blocking alpha presubmit which includes those e2e tests

**Comment by liggitt** [comment]
> you could use `/test pull-kubernetes-e2e-gce-alpha-features` to trigger the non-blocking alpha presubmit which includes those e2e tests

ah, thanks, had trouble finding the pre-submit alpha job name

/test pull-kubernetes-e2e-gce-alpha-features

File: test/e2e/apimachinery/crd_publish_openapi.go [hunk 1]
```diff
@@ -414,7 +415,7 @@ func waitForDefinition(c k8sclientset.Interface, name string, schema []byte) err
 		// drop properties and extension that we added
 		dropDefaults(&d)
 		if !apiequality.Semantic.DeepEqual(expect, d) {
-			return false, fmt.Sprintf("spec.SwaggerProps.Definitions[\"%s\"] not match; expect: %v, actual: %v", name, expect, d)
+			return false, fmt.Sprintf("spec.SwaggerProps.Definitions[\"%s\"] not match;\n%s", name, diff.ObjectReflectDiff(expect, d))
```
Comment (sttts) [inline test/e2e/apimachinery/crd_publish_openapi.go]: does this really work? we have private fields in the `$ref` field if I remember right. The reflect diff complains that it cannot read those. So better double check.

**Comment by sttts** [comment]
The APIVersions changes look good.

File: test/e2e/apimachinery/crd_publish_openapi.go (hunk 1 above)
Comment (liggitt) [reply test/e2e/apimachinery/crd_publish_openapi.go]: hmm, not sure... just trying to get a more useful message than:

> failed to wait for OpenAPI spec validating condition: timed out waiting for the condition; lastMsg: spec.SwaggerProps.Definitions["io.k8s.crd-publish-openapi-test-common-group.v6.E2e-test-crd-publish-openapi-6170-crd"] not match; expect: {{map[]} { {{<nil> {[]} false false false false false}}  Waldo CRD for Testing [object]   <nil> <nil> false <nil> false <nil> <nil>  <nil> <nil> false <nil> [] <nil> <nil> [] <nil> [] [] [] <nil> map[spec:{{map[]} { {{<nil> {[]} false false false false false}}  Specification of Waldo [object]   <nil> <nil> false <nil> false <nil> <nil>  <nil> <nil> false <nil> [] <nil> <nil> [] <nil> [] [] [] <nil> map[dummy:{{map[]} { {{<nil> {[]} false false false false false}}  Dummy property. []   <nil> <nil> false <nil> false <nil> <nil>  <nil> <nil> false <nil> [] <nil> <nil> [] <nil> [] [] [] <nil> map[] <nil> map[] map[] <nil> map[]} { false <nil> <nil> <nil>} map[]}] <nil> map[] map[] <nil> map[]} { false <nil> <nil> <nil>} map[]} status:{{map[]} { {{<nil> {[]} false false false false false}}  Status of Waldo [object]   <nil> <nil> false <nil> false <nil> <nil>  <nil> <nil> false <nil> [] <nil> <nil> [] <nil> [] [] [] <nil> map[bars:{{map[]} { {{<nil> {[]} false false false false false}}  List of Bars and their statuses. []   <nil> <nil> false <nil> false <nil> <nil>  <nil> <nil> false <nil> [] <nil> <nil> [] <nil> [] [] [] <nil> map[] <nil> map[] map[] <nil> map[]} { false <nil> <nil> <nil>} map[]}] <nil> map[] map[] <nil> map[]} { false <nil> <nil> <nil>} map[]}] <nil> map[] map[] <nil> map[]} { false <nil> <nil> <nil>} map[]}, actual: {{map[]} { {{<nil> {[]} false false false false false}}   [object]   <nil> <nil> false <nil> false <nil> <nil>  <nil> <nil> false <nil> [] <nil> <nil> [] <nil> [] [] [] <nil> map[] <nil> map[] map[] <nil> map[]} { false <nil> <nil> <nil>} map[]}

**Comment by liggitt** [comment]
Verified this fixes the CRD conversion alpha tests. The openapi ones are being fixed in a separate PR. Tagging per https://github.com/kubernetes/kubernetes/pull/78190#issuecomment-494551334

**Comment by liggitt** [comment]
looks like CI *really* wants to run the alpha job. will open a new PR

**Pull request closed without merging**